Isc24 (#3)
coderefinery/reports-and-proposals: isc24_tutorial/isc24_tutorial_proposal.md

# ISC '24 tutorial submission

* Deadline: 08.12.23
* ISC - Tutorials: 12.05.24

## Description

### General Information

**Title of the tutorial**: Best Practices in HPC Training
**Tutorial author(s) information**: SW, MJ, DI, TW, RB, ...

### Abstract (Maximum words: 200)

This tutorial will focus on helping competent practitioners and experts teach

> their knowledge to others. It will also serve as a kickstart to teach existing

Reviewer: ```suggestion
their knowledge to learners. It will also serve as a kickstart to teach existing
```

community-developed open-source training lessons on topics in HPC, software
engineering, machine learning, etc. Building on instructor training resources
developed within the CodeRefinery project and The Carpentries, the tutorial
will cover 1) the science and philosophy of teaching; 2) how to teach
interactive, hands-on workshops; 3) how to design interactive, hands-on

> training lessons; and 4) how to effectively organise in-person or online

Reviewer 2: Should it also include hybrid courses here?

Author: good idea, i'll add about hybrid

workshops.

After attending this tutorial, participants should:

- Have increased confidence in teaching interactive workshops in online and in-person settings.
- Be familiar with important pedagogical concepts and scientific findings about how people learn new skills.
- Appreciate the importance of frequent hands-on practice in building skill.
- Understand how learning can be improved by following certain best practices in interactive teaching.

> - Know how to backwards-design training material based on intended learning outcomes.

Reviewer: ```suggestion
- Know how to design training material and exercises based on intended learning outcomes.
```

- Know about important differences between online and in-person training and how inherent challenges in online training can be overcome.

> - Be familiar with a technical approach to collaboratively developing open source lessons based on a template.

Reviewer 2: And how to start something like this for their own learning materials, maybe tips on how to adapt the template to their own needs (could just be in materials, not taught?) 
And also how to serve markdown based materials on GitHub pages (without Jekyll; not everyone seems to be aware that it's a "doable option")

Author: I'll add something about "own materials". But should we mention technical specifics like markdown, gh-pages and Jekyll or keep it general?

Reviewer 2: General is better I guess :+1:

> 

Reviewer: ```suggestion
- Know how to measure the success of the training course by collecting feedback during and after the course.
```

### Detailed description of the tutorial content (Maximum pages: 3)

**Overview and goals of the tutorial (takeaways for the audience)**

This tutorial will cover best practices in developing training material and
delivering technical training workshops. Participants will get acquainted with
key concepts from pedagogical science and the philosophy of teaching, learn
practical methods for teaching interactive and engaging hands-on workshops

> which maximise learning through careful consideration of cognitive load,

Reviewer 2: ```suggestion
which maximises learning through careful consideration of cognitive load,
```

Author: but it's the "practical methods" (plural) that "maximise", not "maximises"

acquire skill in developing effective training lessons that facilitate memory
consolidation, and get practical tips and tricks on how to effectively organise
in-person or online workshops.

After attending this tutorial, participants will:

- Have increased confidence in teaching interactive workshops in online and in-person settings.
- Be familiar with important pedagogical concepts and scientific findings about how people learn new skills.
- Appreciate the importance of frequent hands-on practice in building skill.
- Understand how learning can be improved by following certain best practices in interactive teaching.
- Know how to backwards-design training material based on intended learning outcomes.
- Know about important differences between online and in-person training and how inherent challenges in online training can be overcome.
- Be familiar with a technical approach to collaboratively developing open source lessons based on a template. 

**Targeted audience**

Teaching is a profession, but also something that everyone needs to be able to
do to some degree, since everyone has their own personal specialties they will

> either teach or mentor.

Reviewer 2: Maybe we could add something here about HPC (conference specific), that HPC can have a steep learning curve for "non computational people": not only people teaching HPC need this, but also HPC support and people working with eg infra need to explain design choices to newcomers?

Author: yes, something HPC specific should be added. I'll modify

This course can be relevant for different learner personas:

- **You run a practical teaching program** at your institution (for example as part of a research computing group) and would like to learn best practices for collaborative teaching of technical topics, so that you aren’t re-inventing the same thing over and over again.
- **You are a technical specialist** who is frustrated with the way you currently try to teach others who need to use your software or infrastructure. You can’t spend too much time to become a professional, but you know you need something more than what you’ve been doing. Thus, you would like to adopt some of the best practices of designing and teaching interactive, hands-on workshops.

> - **You’ve been teaching alone**, but would like to join a collaboration network for more co-teaching and to reduce the amount of duplication of effort.

Reviewer 2: ```suggestion
- **You’ve been teaching alone**, but would like to join a internal/national/international collaboration network for more co-teaching and to reduce the amount of duplication of effort.
```
?

- **You are interested in teaching CodeRefinery/ENCCS/... lessons**, and would like a comprehensive kick-start to how these workshops work, either to join and teach with us, or teach our lessons independently.

> **Detailed outline of the tutorial (with time slots)**

Reviewer: Are the breaks pre-set by the conference? Or some of them? I would otherwise suggest to re-balance a bit but maybe there is no wiggle-room.

Author: yes the 30 minute breaks are pre-set by the conference. I prioritised having the breaks after full episodes rather than splitting episodes across a break, but the result is that the 10 minute breaks are not well separated from the 30 minute breaks. I'll see if it can be improved, but maybe this isn't critical and during the workshop we can improvise the shorter breaks

| Time | Topic |
| ---- | ----- |
|09:00-09:30 | Welcome and introduction|
|09:30-09:45 | Training providers - EuroCC, CodeRefinery, Carpentries, ...|
|09:45-10:00 | Available public training resources|
|10:00-10:10 | Break|
|10:10-11:00 | Acquisition of skill|
|11:00-11:30 | Break|
|11:30-12:10 | Lesson design|
|12:10-13:00 | Good interactive teaching practices|
|13:00-14:00 | Lunch break|
|14:00-14:40 | Teaching mechanics|
|14:40-15:20 | Collaborative lesson development|
|15:20-15:30 | Break|
|15:30-16:00 | Practice: teaching and giving feedback|
|16:00-16:30 | Break|
|16:30-17:30 | Practice: teaching and giving feedback|

> |17:30-18:00 | Recap, final discussion and where to go from here|

Reviewer: ```suggestion
|17:30-18:00 | Recap, round-table discussion and where to go from here|
```

> **Description of hands-on activities, if applicable**

Reviewer 2: I wonder if the teaching demonstrations are feasible in this setting where everyone is together in one room, which may possibly be full of people (ie no space to move around)? 
For watching at least we could do so that everyone watches the same clips on screen together and then discuss in smaller groups. For the actual teaching demo not sure. 
What do you think?

Author: i think we'll have to figure that out later and just go for this schedule in the submission. I'm hoping we'll be able to split participants up into groups of 3-4 who can discuss among themselves and also do teaching demos, somehow. But we'll get to know later which room we get

Reviewer 2: At least in SC review, reviewers were looking very detailed in the technical implementation plan, so maybe we could reflect in the text that we are aware that this might be tricky but will adapt according to room assignment and technical possibilities?

The tutorial will include ample time for a variety of hands-on activities.

> Participants be split into groups already during the first morning session

Reviewer 2: ```suggestion
Participants will be split into groups already during the first morning session
```

where they take turns to teach a short snippet of a technical topic to each
other - e.g. the Unix command line - followed by providing feedback to each
other. This experience will form a baseline. In following sessions,
participants will be split into groups of varying sizes to collectively
brainstorm and discuss pedagogical concepts or provide feedback to recorded
teaching demonstrations.  There will also be technical exercises where
participants learn how to use the Sphinx Python package to develop training
lessons in a team using collaborative workflows in Git.  In the final hands-on
activity, participants will again teach a topic to each other in groups and
provide feedback, and reflect on how the concepts learned during the tutorial
have affected their teaching compared to the baseline.

> **Updates from previous presentations, if applicable**

Reviewer 2: Should there be a mention of the CodeRefinery train the trainer courses in previous years as well?

Author: yes! i'll add

This tutorial was taught over two half days to around 30 participants in an
online training workshop targeting the EuroCC network on September 28-29, 2023
(https://enccs.se/events/best-practices-in-online-and-in-person-training/). The
feedback from this workshop was exceptionally positive, with an average rating
of over 9 out of 10 and 100% of participants reporting that they would
recommend the training to their colleagues.  For the ISC tutorial, some of the
exercises will be adapted to the in-person setting, and other minor
improvements will be made to further improve the flow of the training.

**Number of attendees from previous tutorials, if applicable**

Around 30 participants attended the online workshop delivered on September
28-29 2023. This will be the first time the tutorial is taught in-person at
ISC.

**URLs to sample slides and other material**

The training material - which will be somewhat updated before the tutorial -
can be found at https://enccs.github.io/instructor-training/.

## Logistics

**Length of the tutorial: half-day (3.5 hours) or full-day (7 hours)**

full-day 7 hours

**Percentage of content split as beginner/intermediate or intermediate/advanced**

> 50% beginner/intermediate, 50% intermediate/advanced

Reviewer 2: Isn't it all quite beginner friendly? In the context of an HPC conference I would say it is. But of course it depends on the background of the people joining the tutorial.

Author: the beginner/intermediate/advanced categories are much more appropriate for a technical HPC workshop. For this tutorial the technical prerequisites are very limited, but somehow it sends the wrong signal to say that this tutorail is at 100% beginner level. Maybe something like 30/50/20 beginner/intermediate/advanced?

Reviewer 2: True, I guess also the 50/50 is fine as you have it 👍

**Requirements for attendees (e.g., laptop, remote server account)**

- laptop

> - (optional, maybe) Python installation with some libraries, alternatively cloud

Reviewer: Why the Python? Is this for Sphinx?

Author: yes this is for doing the sphinx exercises in https://enccs.github.io/instructor-training/collaborative-development/

but everything can be done in the browser and of course it'll be simpler for us and for the participants to not have to worry about local python, sphinx and git installations. Is that the best way to go? Shall we remove the requirements?

- 30 minute preparation time

**Estimated number of attendees (with short explanation, if applicable)**

> - 40 participants - this way the tutorial can be interactive, inclusive, hands-on... WRITEME

Reviewer 2: ```suggestion
Based on interest in the previously taught versions of this course, we would expect around 30 learners. To keep the tutorial as interactive, inclusive and hands-on as possible, we would like to limit the number of participants to about 40.
```
